GCTC-NTGC/gc-digital-talent
# Debt - 4672 single search request table

Eventually will resolve #4672 

(UPDATED)
## Notes
- Uses the combined efforts of @vd1992 and @tristan-orourke.
- Adds a new input `PoolCandidateSearchInput` for the `poolCandidatesPaginated` query.
```
input PoolCandidateSearchInput {
  applicantFilter: ApplicantFilterInput @spread
  priorityWeight: [Int] @scope(name: "priorityWeight")
  status: [PoolCandidateStatus] @scope(name: "poolCandidateStatuses")
  generalSearch: string
  name: string
  email: string
}
```
- Refactors the `ApplicantFilterInput` and ` PoolCandidateFilterInput` to use the scope directive instead of the builder. A scope directive will cause Lighthouse to look for the appropriatly named scope method on whatever Class its returning, in this case PoolCandidate.
- Adds new test page `PoolCandidateSearchTest.php` 

- @tristan-orourke I think we should ask DCMO what filters they want in the `PoolCandidateFilterDialog` before adding this.
- Question: Should we rename the `PoolCandidateSearchFilter` to something more accurate?

## Testing 
- The best way to test this is to go to the [Search page](http://localhost:8000/en/search).
- Create a new request.
- Go to [admin talent requests page](http://localhost:8000/en/admin/talent-requests) and view our recently created request.
- Ensure number of candidates match total requested.
- Ensure when clicking on Filters dialog that all `initialFilters` are set.
